script.py

```diff
@@ -1,11 +1,207 @@
+                translator (string, optional): Translator assigned to the project.
+                reviewer (string, optional): Reviewer assigned to the project.
+                status (string, optional): Current project status inside the agency's workflow.
+                domain (str): Overall domain to which the project belongs.
+                today (date): Date of the day where the script is run.
+                st (date): Project's start date, turned from an ISO-formatted string into a date.
+                dl (date): Project's deadline, turned from an ISO-formatted string into a date.
+                daysleft (timedelta): Time left until the project's deadline.
+                length (timedelta): Total time allocated to the project.
+                rate (float): Project's word rate.
+                efficiency (float): Number of words to be translated or revised per day to meet the deadline (assuming a five-day workweek).
+            
+            Raises:
+                TypeError: If 'title' is not a string.
+                TypeError: If 'client' is not a string.
+                TypeError: If 'source' is not a string.
+                TypeError: If 'target' is not a string.
+                TypeError: If 'words' is not an integer.
+                TypeError: If 'start' is not a string.
+                ValueError: If 'start' does not follow the pattern 4 digits-2 digits-2 digits.
+                TypeError: If 'deadline' is not a string.
+                ValueError: If 'deadline' does not follow the pattern 4 digits-2 digits-2 digits.
+                TypeError: If 'price' is not a float.
+                TypeError: If 'tm' is not a boolean.
+                TypeError: If 'translator' is not a string.
+                TypeError: If 'reviewer' is not a string.
+                TypeError: If 'status' is not a string.
+                ValueError: If 'status' is not a label in the agency workflow: created, in translation, in revision, delivered, delayed, cancel(l)ed.
+                TypeError: If 'domain' is not a string.
+        """
+        if type(title) != str:
+            raise TypeError("The title must be a string.")
+        else:
+            self.title = title
+        if type(client) != str:
+            raise TypeError("The client name must be a string.")
+        else:
+            self.client = client
+        if type(source) != str:
+            raise TypeError("The source language must be a string.")
+        else:
+            self.source = source
+        if type(target) != str:
+            raise TypeError("The target langague must be a string.")
+        else:
+            self.target = target
+        if type(words) != int:
+            raise TypeError("The word count must be an integer.")
+        else:
+            self.words = words
+        if type(start) != str:
+            raise TypeError("The start date must be provided as a string.")
+        elif not re.match ("[0-9]{4}-[0-9]{2}-[0-9]{2}", start):
```

Comment:
For validation of your ISO Format dates: the regex is not super robust (you could have 0000-00-00 and it would match, or 0123-40-54...), so I would try the following instead:

```python
try:
    self.start = datetime.date.fromisoformat(start)
except:
   raise TypeError("The start date must be provided in ISOFormat")
```

This has the advantage of also already converting your date :)

Reply:
I changed it, but the problem is that then, when I call the attribute "deadline", I once again have the problem that it's fairly "cumbersome"-looking instead of simply displaying the ISO-format date (the string). So, it's a more efficient code, but it's not very user-friendly in attribute display.

Reply:
This is similar to what I've mentioned before of having private and public attributes. You could have `_start` and `_deadline` (or `st` and `dl`) as the private versions used in computing other things, and just `start` and `deadline` for printing. Or you could have a method that returns the isoformat version of a date.

In any case, according to [the docs](https://docs.python.org/3/library/datetime.html#datetime.date.isoformat), if you have a value of class `date`, you can turn them back into isoformat with `.isoformat()` or just print them. That is, even if `start` is a `date`, `my_project.start.isoformat()` will print it nicely and `print(my_project.start)` (instead of just `my_project.start`) will _also_ print it nicely.